### PR TITLE
Marketplace: Create a component to display reviews

### PR DESCRIPTION
**Pull request by WBerredo**
Related to https://github.com/Automattic/dotcom-forge/issues/4562

## Proposed Changes

Create a component that receives product slug and product type and display all the reviews for this product.

| Desktop  | Mobile |
| ------------- | ------------- |
|CleanShot 2023-11-29 at 14 04 26@2x|CleanShot 2023-11-29 at 14 05 55@2x|

### Out of scope
* Pagination will be handled  as a follow-up
* The renaming of the types from plugin to product are handled on https://github.com/Automattic/wp-calypso/pull/84480/

## Testing Instructions
* Add to the `plugin-details.jsx` file
* On the import area 
```ts
import { MarketplaceReviewsList } from '../marketplace/components/reviews-list';
```

* On line `454`
```ts
<MarketplaceReviewsList slug={ fullPlugin.slug } productType="plugin" />
```
* Go to a plugin page with reviews. Ex: `plugins/woocommerce-bookings`
* Go to the bottom of the page and check if the reviews are displayed there
* Also test on the mobile view
* Also check plugins without a reviews. Ex: `/plugins/wordpress-seo-premium`

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/marketplace/components/reviews-list/index.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	MarketplaceReviewResponse,
+} from 'calypso/data/marketplace/use-marketplace-reviews';
+import './style.scss';
+import { sanitizeSectionContent } from 'calypso/lib/plugins/sanitize-section-content';
+
+export const MarketplaceReviewsList = ( props: ProductProps ) => {
+	const translate = useTranslate();
+	const { data: reviews } = useMarketplaceReviewsQuery( props );
+
+	if ( ! isEnabled( 'marketplace-reviews-show' ) ) {
```
Comment (cpapazoglou) [inline client/my-sites/marketplace/components/reviews-list/index.tsx:17]: nitpick: this can be on the top

**Review by cpapazoglou — CHANGES_REQUESTED** [review]
Let's handle mobile view better. 

CleanShot 2023-11-29 at 12 56 58@2x

CleanShot 2023-11-29 at 12 57 15@2x

I think we would be OK on this, even without designs

CleanShot 2023-11-29 at 12 58 34@2x

**Review by epeicher — APPROVED** [review]
I have tested it, and it works as advertised. I have checked normal and mobile views for plugins with and without reviews and the behaviour is as expected, also the design looks nice, thanks for adding some design touches to this initial version 😄 

I would additionally create follow-up design tasks for the mobile version as in mobile view, the date column dissapppears and in widths of around 440px, the date column is truncated. I guess that can be completed when we have some design input. I would take some inspiration from the battle tested Woo reviews component [example here](https://woo.com/products/woocommerce-bookings/?quid=fd1fe2ceb16e92acac2157d05ebe7945#reviews) that collapses the columns when reducing the width.



Overall, it Looks Good to Me and I would :shipit: 

EDIT: I didn't see that Harris reviewed it first so maybe it's worth having further discussions 😄

File: client/my-sites/marketplace/components/reviews-list/index.tsx (hunk 1 above)
Comment (epeicher) [reply client/my-sites/marketplace/components/reviews-list/index.tsx:17]: As this is a condition, it cannot be added before calling the hooks (`useTranslate` and `useMarketplaceReviewsQuery`)  as they will be conditionally called and that breaks the hook's rules 😄

Comment (cpapazoglou) [reply client/my-sites/marketplace/components/reviews-list/index.tsx:17]: This happens when you are not writing code for a long time 🤦 . Thanks for the kind explanation Roberto.

**Comment by WBerredo** [comment]
Mobile layout updated

**Review by cpapazoglou — APPROVED** [review]
Thanks for the quick turnaround. LGTM!

**Review by epeicher — APPROVED** [review]
Thanks @WBerredo for addressing the comments. The mobile view now looks great 🎉

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9795435

Some locales (Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @WBerredo for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9795435

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @WBerredo for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9795435).

**Pull request merged**